text_generation/causal_lm/cpp/chat_model_lm.cpp

```diff
@@ -115,7 +117,10 @@ int main(int argc, char* argv[]) try {
                 }
             }
         }
-        std::cout << "Answer: " << detokenize(detokenizer, answer.tokens) << "\n_______\n";
+
+        auto answer_str = detokenize(detokenizer, answer.tokens);
+        answer_str = answer_str.substr(0, answer_str.find("<eos>"));
```

Comment:
why do we need this hack?
CC @apaniukov

Reply:
Due to beam search. Some beams may be much shorter than others and they looks like this 
"The answer is 4. 2 + 2 = 4.<eos><eos><eos><eos><eos><eos><eos><eos>..."
I add this just to make output more readable

Reply:
No, adding `--skip-special-tokens` to the `convert_tokenizer` command should remove these tokens during detokenization.

Why don't we end the beam generation in the case of the `<eos>` token anyway? It should speed up other beam generation in that case. It also might affect a beam score.

Reply:
> Why don't we end the beam generation in the case of the <eos> token anyway?

It should work this way, I wonder why EOS does not work..

Reply:
It should indeed, I'll check this sample

Reply:
I cannot reproduce it locally. Getting result without eos tokens from detokenizer. I'll sync with @olpipi on that.